src/MonoTorrent/MonoTorrent.Client/Modes/StoppingMode.cs

```
@@ -77,6 +78,8 @@ public async Task WaitForStoppingToComplete (TimeSpan timeout)
                 var overallTasks = Task.WhenAll (stoppingTasks);
                 if (await Task.WhenAny (overallTasks, delayTask) == delayTask)
                     Logger.Log (null, "Timed out waiting for the announce request to complete");
+                else
+                    await overallTasks;
```

Add a test and let the exception propagate?